src/components/Appointments/AppointmentSlot.vue

```diff
@@ -0,0 +1,72 @@
+<!--
+  - @copyright 2021 Christoph Wurst <christoph@winzerhof-wurst.at>
+  -
+  - @author 2021 Christoph Wurst <christoph@winzerhof-wurst.at>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -->
+
+<template>
+	<button class="appointment-slot">
+		{{ startTime }} - {{ endTime }}
+	</button>
+</template>
+
+<script>
+export default {
+	name: 'AppointmentSlot',
+	props: {
+		start: {
+			required: true,
+			type: Number,
+		},
+		end: {
+			required: true,
+			type: Number,
+		},
+		timeZoneId: {
+			required: true,
+			type: String,
+		},
+	},
+	computed: {
+		dateTimeFormatter() {
+			// TODO: detect locale as well
+			return Intl.DateTimeFormat('en-US', {
+				timeZone: this.timeZoneId,
+				timeStyle: 'full',
+				dateStyle: 'short',
+			})
+		},
+		startTime() {
+			// TODO: is there a better way to determine the locale? The one from @nextcloud/l10n is not always supported
```

Comment:
Supported by?

Reply:
`Date.toLocaleString`. It doesn't work with `de_AT` for example

Reply:
That's because it should be `de-AT`. :)

Reply:
Yeah, figured. But then we have to convert and I'm not sure if replacing `_` with `-` works for all possibly locales

Reply:
That's pretty much what we do in https://github.com/nextcloud/calendar/blob/71be054f7d58556d164bfdd0d72551a8a98ec42e/src/utils/moment.js#L34-L35

But that's also indeed what cause trouble when Nextcloud locales aren't supported by fullcalendar.

Reply:
I discovered `getCanonicalLocale` that does exactly that and added some try-catch around to fall back to `undefined`. This *should* work universally.